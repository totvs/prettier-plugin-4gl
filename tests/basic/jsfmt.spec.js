//run_spec(__dirname, ['4gl'], { '4glKeywordsCase': 'ignore' });
run_spec(__dirname, ['4gl'], { '4glKeywordsCase': 'upper' });
//run_spec(__dirname, ['4gl'], { '4glKeywordsCase': 'lower' });
