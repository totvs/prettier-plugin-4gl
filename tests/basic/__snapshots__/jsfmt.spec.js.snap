// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source builtInVars.4gl 1`] = `
main

    let x, y = arg_val
    let x, y = arr_count
    let x, y = arr_cur
    let x, y = sqlca
    
end main
----------
MAIN

  LET x, y = arg_val
      LET x, y = arr_count
      LET x, y = arr_cur
      LET x, y = sqlca


END MAIN



`;

exports[`Source empty_main.4gl 1`] = `
main

end main
----------
MAIN


END MAIN



`;

exports[`Source globals.4gl 1`] = `
globals "abc.txt"

  globals
    DEFINE NOME CHAR(50) # abc
      DEFINE NOME CHAR(50) # abc
        DEFINE NOME CHAR(50) # abc

    DEFINE NOME2 CHAR(50) # abc
end globals




main
call xxxxx()



call xxxxx()



call xxxxx()



call xxxxx()
end main
----------
GLOBALS "abc.txt"

GLOBALS
  DEFINE NOME CHAR(50) # abc
  DEFINE NOME CHAR(50) # abc
  DEFINE NOME CHAR(50) # abc

  DEFINE NOME2 CHAR(50) # abc

END GLOBALS




MAIN
  CALL xxxxx()



  CALL xxxxx()



  CALL xxxxx()



  CALL xxxxx()

END MAIN



`;

exports[`Source strings.4gl 1`] = `
main

    let x, y = ''
    let x, y = 'ABC,DEF'
    let x, y = '"'
    let x, y = 'ABC\\'DEF'
    
end main

function f1()

    let x, y = ""
    let x, y = "ABC,DEF"
    let x, y = "'"
    let x, y = "ABC\\"DEF"
    
end function
----------
MAIN

  LET x, y = ''
      LET x, y = 'ABC,DEF'
      LET x, y = '"'
      LET x, y = 'ABC\\'DEF'


END MAIN

FUNCTION f1()

  LET x, y = ""
      LET x, y = "ABC,DEF"
      LET x, y = "'"
      LET x, y = "ABC\\"DEF"


END FUNCTION



`;
